### ISSUE TITLE
[Feature]: support for NEF / raw files

### ISSUE THREAD

**Issue by ymarkus** [immich-app/CLI#47]
As Immich supports NEF files, but right now there's no bulk upload...

**Comment by alextran1502** [comment]
What do you mean by this? Can you elaborate?

**Comment by ymarkus** [comment]
Oh, I just realized it is supposed to support it, sorry!
For some reason, it doesn't find the files in my directory. I'll upload some screenshots:
![2022-12-26T23-33-27](https://user-images.githubusercontent.com/62380378/209586675-46fb0e22-a43f-4d91-9eff-1fcf56eeab96.png)
![2022-12-26T23-32-03](https://user-images.githubusercontent.com/62380378/209586682-dd88b532-b3a2-4303-8b9e-8a47c6867e25.png)
![2022-12-26T23-34-53](https://user-images.githubusercontent.com/62380378/209586723-7de9b3bf-b43a-4664-b590-4234692a5a45.png)

**Comment by ymarkus** [comment]
Ok, I think I have found my issue:
```
❯ file --mime-type Documents/raw/2020/Island/100D5000/DSC_0001.NEF
Documents/raw/2020/Island/100D5000/DSC_0001.NEF: image/tiff
```
any idea why it says tiff and not nef?

**Comment by ymarkus** [comment]
Ok, I think I've found the issue:
It seems like this CLI looks for these mime types: `"image/nef", "image/x-nikon-nef",` 
These are not standard mime types according to https://www.iana.org/assignments/media-types/media-types.xhtml and are also not "custom filetypes" in [mime-db](https://github.com/jshttp/mime-db/blob/master/src/custom-types.json).
Did I miss something or do I need to open a PR there to get it registered as "custom filetype"?

**Comment by alextran1502** [comment]
Is there an example file that you can share for testing? I can update the file mime type once I got the correct one

**Comment by ymarkus** [comment]
Yes, of course. I'll send it by e-mail if it's ok for you.
So, I've been having a conversation at https://github.com/jshttp/mime-db/pull/219 with the maintainer there. This is the mime-db package `mime-types` depends on as explained in https://www.npmjs.com/package/mime-types. It seems like they are not able to add those mime types to their database. Maybe we need a different way of recognizing mime types / file types.
Another indication that "nef" mime types are hit or miss:
```
❯ mimetype raw/2020/Island/100D5000/DSC_0001.NEF
raw/2020/Island/100D5000/DSC_0001.NEF: image/x-nikon-nef
❯ file --mime-type raw/2020/Island/100D5000/DSC_0001.NEF
raw/2020/Island/100D5000/DSC_0001.NEF: image/tiff
```

**Comment by ymarkus** [comment]
Ok, so I did some more investigation. The mime-types package returns `false` as mimeType for all NEF or nef files. Do you think checking for filename extensions would be fine for those mime types missing in mime-db?

**Comment by n3u1r0n** [comment]
I do have the same issue with dng files. The mime-types package simply returns `false` and they do not get uploaded.

**Comment by n3u1r0n** [comment]
As a workaround (obviously not a good solution)  and a lead to where the bug lies:
I commented out the mime-type check in the CLI
```
// if (SUPPORTED_MIME.includes(mimeType)) {
    if (mimeType === false) {
        mimeType = "image/x-adobe-dng";
    }
    const fileStat = fs.statSync(filePath);
    localAssets.push({
        id: `${path.basename(filePath)}-${fileStat.size}`.replace(/\s+/g, ""),
        filePath,
    });
// }
```
and adapted the `getAssetType` function in the CLI
```
function getAssetType(filePath) {
    var mimeType = mime.lookup(filePath);
    if (mimeType === false) {
        mimeType = "image/x-adobe-dng";
    }
    return mimeType.split("/")[0].toUpperCase();
}
```
which gave me the 400 response above.

So so I went inside the docker container of the Immich server I commented out the mime-type check in `/usr/src/app/dist/apps/immich/apps/immich/src/config/asset-upload.config.js`
```
//if (file.mimetype.match(/\/(jpg|jpeg|png|gif|mp4|webm|x-msvideo|quicktime|heic|heif|dng|x-adobe-dng|webp|tiff|3gpp|nef|x-nikon-nef)$/)) {
    cb(null, true);
//}
//else {
//    logger.error(`Unsupported file type ${(0, path_1.extname)(file.originalname)} file MIME type ${file.mimetype}`);
//    cb(new common_1.BadRequestException(`Unsupported file type ${(0, path_1.extname)(file.originalname)}`), false);
//}
```

When I now upload my `.dng`'s everything works fine.

**Comment by uhthomas** [comment]
Yeah, I observed this too. The cli tool will need some more custom mime type stuff given the upstream library doesn't have an api to add custom mime types(!?). I'll probably take a look at this soon.

**Comment by kmaid** [comment]
Just chiming in here that I have noticed that NEFS were not added when imported via an external library
